tools/wptrunner/wptrunner/environment.py

```diff
@@ -50,6 +51,76 @@ class TestEnvironmentError(Exception):
     pass
 
 
+class _QueueLogger(StructuredLogger):
+    """Structured logger that has a unique name and puts messages on a queue"""
+    def __init__(self, queue):
+        name = str(uuid.uuid4())
+        self.queue = queue
+        super().__init__(name)
+        assert not self.handlers
+
+    def _handle_log(self, data):
```

Comment:
What ends up calling `_handle_log`? Something in mozlog?

Reply:
Yes, https://searchfox.org/mozilla-central/source/testing/mozbase/mozlog/mozlog/structuredlog.py#278